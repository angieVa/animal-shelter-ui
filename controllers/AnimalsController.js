import axios from 'axios';

export const AnimalController = {
    register(animal) {
        return axios({
            url: `animals`,
            baseURL: process.env.API,
            data: animal,
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
              },
        })
    },
    list() {
        return axios({
            method: 'GET',
            baseURL: process.env.API,
            url: 'animals'
        });
    },
    delete(name) {
        return axios({
            method: 'DELETE',
            baseURL: process.env.API,
            url: `animals/${name}`,
        });
    },
}

