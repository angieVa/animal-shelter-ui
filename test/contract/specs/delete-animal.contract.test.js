import { provider } from "./init-pact";
import { AnimalController } from "../../../controllers";
import { Matchers } from "@pact-foundation/pact";

const name = "Jasper";

describe('Given An Animal service', () => {
    describe('When a request to delete an animal is made', () => {
        beforeAll(async () => {
            await provider.setup();
            await provider.addInteraction({
                state: 'name animal',
                uponReceiving: 'a request to delete an animal',
                withRequest: {
                    method: 'DELETE',
                    path: `/animals/${name}`,
                },
                willRespondWith: {
                    status: 204,
                }
            });
        });

        it("Then it should return the right status", async() =>{

            const response = await AnimalController.delete(name);
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        });

        afterAll(async () => {
            await provider.finalize();
        });
    });
}); 