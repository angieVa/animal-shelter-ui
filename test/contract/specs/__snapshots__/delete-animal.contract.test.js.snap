// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given An Animal service When a request to delete an animal is made Then it should return the right status 1`] = `""`;
